File: Security/caesar_cipher/caesar_cipher.go
package main

import (
    "fmt"
    "strings"
)

func main() {
    alphabet := [] string {"a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"}
  
    fmt.Println("Please enter the mode you want...")
    var mode string
    fmt.Scan(&mode)
    mode = strings.ToLower(mode)
  
    if mode == "encrypt" {
      fmt.Println("Please enter the message you want encrypted...")
      var message string
      fmt.Scan(&message)
      
      fmt.Println("Please enter the shift number...")
      var shift int
      fmt.Scan(&shift)
      
      var newmsg string

      for i := range message {
        
        letter := string(message[i])
        var index int
        
        for ai := range alphabet {
          if alphabet[ai] == letter {
            index = ai
          }
        }
        
        if (index + shift) > len(alphabet) {
          for (index + shift) > len(alphabet) {
            shift -= len(alphabet)
          }
        }
        newmsg += alphabet[index + shift]
      }
      fmt.Println(newmsg)
    } else if mode == "decrypt" {
      fmt.Println("Please enter the message you want decrypted...")
      var message string
      fmt.Scan(&message)
      
      fmt.Println("Please enter the shift number...")
      var shift int
      fmt.Scan(&shift)
      
      var newmsg string

      for i := range message {
        
        letter := string(message[i])
        var index int
        
        for ai := range alphabet {
          if alphabet[ai] == letter {
            index = ai
          }
        }
        
        if (index + shift) > len(alphabet) {
          for (index + shift) > len(alphabet) {
            shift -= len(alphabet)
          }
        }
        
        newmsg += alphabet[index - shift]
      }
      fmt.Println(newmsg)
    }
}

File: Security/caesar_cipher/caesar_cipher.cpp
#include <iostream>

int main(void) {
        std::string type;
        std::cout << "Would you like to encrypt or decrypt a message?" << std::endl;
        std::cin >> type;
        bool encrypt = type == "encrypt";
        std::string message;
        std::cout << "Enter the message you want " << (encrypt? "encrypted" : "decrypted") << "..." << std::endl;
        std::cin >> message;
        int shift;
        std::cout << "Enter the shift number..." << std::endl;
        std::cin >> shift;
        std::string a_lower = "abcdefghijklmnopqrstuvwxyz", a_upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        std::string output;
        shift %= 26;
        if (!encrypt)
                shift *= -1;
        for (char c: message) {
                int i = a_lower.find(c, 0), j = a_upper.find(c, 0);
                if (i != -1)
                        output += a_lower[(i + shift + 26) % 26];
                else if (j != -1)
                        output += a_upper[(i + shift + 26) % 26];
                else
                        output += c;
        }
  
        std::cout << output << std::endl;
        return 0;
}
